modules/l4proxyprotocol/handler.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Matthew Holt
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4proxyprotocol
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/mastercactapus/proxyprotocol"
+	"github.com/mholt/caddy-l4/layer4"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(Handler{})
+}
+
+// Handler is a connection handler that accepts the PROXY protocol.
+type Handler struct {
+	// How long to wait for the PROXY protocol header to be received (default 5s).
+	Timeout caddy.Duration `json:"timeout,omitempty"`
+
+	logger *zap.Logger
+}
```

Comment:
I don't see an `allow` list of incoming IP addresses here. I think this should be added, to match the Caddy plugin https://github.com/mastercactapus/caddy2-proxyprotocol, and because it's an important security constraint to protect against untrusted clients stuffing the request with a bogus remote IP, to spoof the request etc.